Chore: remove unused feature flag showFeatureFlagsInUI

ryantxu opened the pull request:
Very early versions of feature flags had an option to show them in the UI -- that does not exist so we should remove the flag

ryantxu committed:
remove unused feature flag

ArturWierzbicki reviewed (approved):
there is still bit to remove (together with handler)

https://github.com/grafana/grafana/blob/6c5a5737721914cdce87bf27990a89e0fe6d47cb/pkg/api/api.go#L636-L638

ryantxu committed:
Merge remote-tracking branch 'origin/main' into remove-showFeatureFlagsInUI

ryantxu committed:
remove handler

ryantxu committed:
Live: Remove dev code that would save config and messages in SQL (#62912)

ryantxu committed:
Merge remote-tracking branch 'origin/main' into remove-showFeatureFlagsInUI

ryantxu committed:
merge main

ryantxu committed:
merge main

ryantxu committed:
keep the getter it is used

ryantxu committed:
ok not this one

ryantxu committed:
keep old function

ryantxu commented on pkg/services/featuremgmt/manager.go:
```diff
@@ -152,17 +151,8 @@ func (fm *FeatureManager) GetFlags() []FeatureFlag {
-
-	response.JSON(http.StatusOK, res).WriteTo(c)
+	// pkg/services/accesscontrol/scope_test.go
+	response.Err(fmt.Errorf("WHY DOES REMOVING THIS BREAK TESTS!!!!"))
```
@ArturWierzbicki -- I am stumped, removing this function has my local + drone fail when running tests with:
```
# github.com/grafana/grafana/pkg/services/accesscontrol.test
/opt/homebrew/Cellar/go/1.19.5/libexec/pkg/tool/darwin_arm64/link: running clang failed: exit status 1
Undefined symbols for architecture arm64:
  "_github.com/grafana/grafana/pkg/api/response.wrap_handler", referenced from:
      _github.com/grafana/grafana/pkg/web.mwFromHandler in go.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

FAIL	github.com/grafana/grafana/pkg/services/accesscontrol [build failed]
FAIL
```

DanCech replied:
that is very weird indeed, I wonder if it could be related to removing the import of `github.com/grafana/grafana/pkg/api/response` rather than the function itself, but I haven't been able to figure out how that might impact anything either

DanCech replied:
That seems to have done the trick :)

alexanderzobnin committed:
SAML: Update library (fix single logout) (#62880)

* SAML: update library (fix single logout)

* Run go mod tidy

andresmgot committed:
Prometheus: Refactor. Use data source UID rather than the deprecated ID (#62795)

renovate[bot] committed:
Update dependency eslint to v8.33.0 (#62913)

* Update dependency eslint to v8.33.0

* update eslint sdk as well

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Ashley Harrison <ashley.harrison@grafana.com>

andresmgot committed:
Refactor: Use data source UID rather than ID (#62781)

joshhunt committed:
NestedFolders: use guardian.NewByUID to check folder permissions (#62864)

matyax committed:
Loki Query Editor: Make Monaco the default editor (#62247)

* feat(loki-editor): remove slate and make Monaco the default editor

* feat(loki-editor): remove unsupported usages of onBlur

* feat(loki-editor): remove monaco feature flag

* Chore: remove unused import

leventebalogh committed:
Connections: Fix redirection when creating new DS (#62938)

fix: redirect to connections when creating a new DS from the catalog

papagian committed:
Chore: Fix logging message in folder service (#62939)

leventebalogh committed:
Datasources: Fix tracking event name (#62940)

fix: fix typo in event name

bergquist committed:
remote cache: new function to get/set cache items as byte arrays (#62916)

Signed-off-by: bergquist <carl.bergquist@gmail.com>

kalleep committed:
AuthN: support sync cache for proxy client (#62874)

* AuthN: Add cache support for auth proxy to skip sync

* AuthN: Change proxy auth hook to be a client hook

* AuthN: fix cache key

* fix test

* lint

bergquist committed:
Auth: removes temporary cache of user session token (#62730)

Signed-off-by: bergquist <carl.bergquist@gmail.com>

leventebalogh committed:
Datasources: Add optional tracking for datasources list viewed (#62949)

chore: add optional tracking for datasources list viewed

zoltanbedi committed:
MSSQL/Postgres: List views in table dropdown as well (#62867)

* MSSQL: Query views for table dropdown

* Postgres: Query views for table dropdown

gillesdemey committed:
Alerting: fix "policy" typo (#62953)

dsotirakis committed:
CI: Remove variants arg from `package` step (#62858)

* Remove variants arg from package step

# Conflicts:
#	.drone.yml

* Fix starlark lint

* Default to releaseModeConfig.Variants

* Check for empty variants arg

chri2547 committed:
Docs: corrects typo (#62954)

corrects typo

itsgareth committed:
Loki: Add placeholder to the loki query editor (#62773)

* fix: add placeholder to loki query editor

* fix: use original placeholder

* refactor: extract set placeholder into seperate function

* fix: use correct monaco types

* revert: unwanted change from merge

Jguer committed:
Authn: Stat registration (#62934)

* reorganize auth usage stats

* usage stat privilege elevators

* stat count of modified role

* cfg related info

* add authn anon client

* kv store

* ensure anon enabled is collected even if client is not registered

* fix usage stats test

imatwawana committed:
docs: fix broken annotations link (#62834)

* docs: fix broken annotations link

Link to annotations section is broken because the section was never added. A placeholder for it was removed on this commit, but the link wasn't. This seems like the best place to point to.

* fix relref formatting

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

* removed TODOs from tables

* prettier linting

---------

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

Jguer committed:
Support bundles:  Refactor registry into separate service (#62945)

* add bundle registry service to avoid dependency cycles

* move user support bundle collector to user service

* move usage stat bundle implementation to usage stats

* add info for background service

* fix remaining imports

* whitespace

DanCech committed:
try to resolve linker error

leventebalogh reviewed (approved):
LGTM 👍

ryantxu committed:
merge main
